### PR TITLE
install: totally rewritten on data persistting

### PR DESCRIPTION
**Pull request by h404bi**
**still WIP don't merge, needs code review and somebody to test**.
Follow up #2890 

To participate in the code review and tests, go visit https://github.com/h404bi/scoop-persist-test for test cases.

### CODE REVIEW

**Comment by Ash258** [comment]
Should be #2779 closed, merged or ... Do you need

**Comment by h404bi** [comment]
@Ash258 It maybe should be closed, the data persisting logic in pr #2779 is still confusing. Though I have tested for myself for many test cases, I need someone to help me to review and test this pull-request. 

cc @PakornVl @dzxs @mpro7 who have left comments in related issues/pr.

File: lib/install.ps1 [hunk 1]
```diff
@@ -1222,33 +1222,33 @@ function persist_data($manifest, $original_dir, $persist_dir) {
...
-                    # if there is no source and it's a directory we create an empty directory
-                    ensure $target.FullName | out-null
+            $source = fullpath "$dir\$source"
+            $target = fullpath "$persist_dir\$target"
+
+            # if we have had persist data in the store, just create link and go
+            if (Test-Path $target) {
+                # if there is also a source data, rename it (to keep a original backup)
+                if (Test-Path $source) {
+                    Move-Item -Force $source "$source.original"
```
Comment (h404bi) [inline lib/install.ps1]: Should we use `$source.original` or `"$source.bak"`?

**Comment by h404bi** [comment]
This pull-request doesn't resolve #2724 (#2828 ), I'll take a look at it and try to fix it at the same time.

**Pull request closed without merging**